app/Transcripts.tsx

```diff
@@ -0,0 +1,161 @@
+          setVideos(videoData);
+        }
+      });
+
+      return () => {
+        ignore = true;
+      }
+  }, [filters])
+
+  function handleFilterChange(filters: TranscriptFilterSelection) {
+    updateFilters(filters);
+    updateFilterParams(filters);
+  }
+
+  const videoLinks: React.ReactNode[] = videos.map(
+    video => (
+      <li key={video.videoId} className="mx-3 list-disc">
+        <Link href={getVideoPath(filters.category, video.videoId)}>
+          {video.title}
+        </Link>
+      </li>
+    ));
+
+  // Get access to the tailwind theme colors to pass to the loading spinner
+  // Specifying only content and theme keys is a workaround for a type error pertaining
+  // to the "future" key
+  const fullConfig = resolveConfig({ content, theme });
+  const loadingSection = <section className="my-4 flex flex-row">
+    <TailSpin wrapperClass="justify-center" color={fullConfig.theme.colors.blue['500']} />
+  </section>;
+
+  const resultsSection: React.ReactNode = <section>
+    <h2 className="my-4 text-lg">
+      Transcripts:
+    </h2>
+    <ul className="flex flex-col flex-wrap h-screen">
+      {videoLinks}
+    </ul>
+  </section>;
+
+  return (
+    <>
+      <TranscriptFilter selection={filters} onFilterChange={handleFilterChange} />
+      {isLoading ? loadingSection : resultsSection}
+    </>
+  );
+}
+
+function useFilterParams(allCategories: string[]): {
+   filterParams: TranscriptFilterSelection,
+   updateFilterParams: (newFilters: TranscriptFilterSelection) => void
+} {
+  const searchParams: URLSearchParams = useSearchParams();
+  const pathName: string = usePathname();
+  const { push } = useRouter();
+
+  const filterParams: TranscriptFilterSelection = {
+    category: getCategoryFromParams(searchParams, allCategories),
+    dateRange: getDateRangeFromParams(searchParams),
+  };
+
+  const updateFilterParams = (newFilters: TranscriptFilterSelection) => {
+    push(buildUrl(pathName, newFilters), { scroll: false });
+  }
+
+  return {
+    filterParams,
+    updateFilterParams
+  };
+}
+
+function getCategoryFromParams(params: URLSearchParams, allCategories: string[]): string {
+  const categoryParam: string | null = params.get('category');
+
+  if (categoryParam !== null && allCategories.includes(categoryParam)) {
+    return categoryParam;
+  }
+  else {
+    return 'sps-board';
+  }
+}
+
+function getDateRangeFromParams(params: URLSearchParams): DateRange {
+  const start: Date | null = getDateFromParams(params, 'start');
+  const end: Date | null = getDateFromParams(params, 'end');
+
+  if (start === null && end === null) {
+    return { start: parseDateFromPath('2024-03-01'), end: null };
+  }
+  else {
+    return { start, end };
+  }
+}
+
+function getDateFromParams(params: URLSearchParams, key: string): Date | null {
+  const param: string | null = params.get(key);
+  if (param === null) {
+    return null;
+  }
+
+  const date: Date = parseDateFromPath(param);
+
+  if (isValid(date)) {
```

Comment:
nit: prefer early return null on error and last line being the success when easy.